src/ImageSharp/Processing/Processors/Transforms/RotateProcessor.cs

```diff
@@ -17,95 +16,72 @@ namespace SixLabors.ImageSharp.Processing.Processors
-                    0,
-                    height,
-                    configuration.ParallelOptions,
-                    y =>
-                    {
-                        Span<TPixel> targetRow = targetPixels.GetRowSpan(y);
-
-                        for (int x = 0; x < width; x++)
-                        {
-                            var transformedPoint = Point.Rotate(new Point(x, y), matrix);
-
-                            if (sourceBounds.Contains(transformedPoint.X, transformedPoint.Y))
-                            {
-                                targetRow[x] = source[transformedPoint.X, transformedPoint.Y];
-                            }
-                        }
-                    });
-
-                source.SwapPixelsBuffers(targetPixels);
-            }
+            // Tansforms are inverted else the output is the opposite of the expected.
+            Matrix3x2 matrix = Matrix3x2Extensions.CreateRotationDegrees(this.Degrees, PointF.Empty);
+            Matrix3x2.Invert(matrix, out matrix);
+            return matrix;
         }
 
         /// <inheritdoc/>
-        protected override void BeforeApply(ImageFrame<TPixel> source, Rectangle sourceRectangle, Configuration configuration)
+        protected override void OnApply(ImageFrame<TPixel> source, ImageFrame<TPixel> destination, Rectangle sourceRectangle, Configuration configuration)
         {
-            if (MathF.Abs(this.Angle) < Constants.Epsilon || MathF.Abs(this.Angle - 90) < Constants.Epsilon || MathF.Abs(this.Angle - 180) < Constants.Epsilon || MathF.Abs(this.Angle - 270) < Constants.Epsilon)
+            if (this.OptimizedApply(source, destination, configuration))
             {
                 return;
             }
 
-            this.processMatrix = Matrix3x2Extensions.CreateRotationDegrees(-this.Angle, new Point(0, 0));
-            if (this.Expand)
-            {
-                this.CreateNewCanvas(sourceRectangle, this.processMatrix);
-            }
+            base.OnApply(source, destination, sourceRectangle, configuration);
         }
 
         /// <inheritdoc/>
-        protected override void AfterImageApply(Image<TPixel> source, Rectangle sourceRectangle)
+        protected override void AfterImageApply(Image<TPixel> source, Image<TPixel> destination, Rectangle sourceRectangle)
         {
-            ExifProfile profile = source.MetaData.ExifProfile;
+            ExifProfile profile = destination.MetaData.ExifProfile;
             if (profile == null)
             {
                 return;
             }
 
-            if (MathF.Abs(this.Angle) < Constants.Epsilon)
+            if (MathF.Abs(this.Degrees) < Constants.Epsilon)
             {
                 // No need to do anything so return.
                 return;
             }
 
             profile.RemoveValue(ExifTag.Orientation);
 
-            if (this.Expand && profile.GetValue(ExifTag.PixelXDimension) != null)
+            if (profile.GetValue(ExifTag.PixelXDimension) != null)
```

Comment:
I wonder if this will be correct for auto rotated images. This had `this.Expand` set to false which would mean that we would not update the values. But I do think we could update the values though. Not sure why we didn't?

Reply:
Previously the rotations that fell under that category (90, 180, 270) actually ignored the expanded value and automatically expanded anyway. Not updating them was a mistake on my part.